packages/flutter_test/lib/src/binding.dart

```diff
@@ -611,6 +624,20 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   late StackTraceDemangler _oldStackTraceDemangler;
   FlutterErrorDetails? _pendingExceptionDetails;
 
+  MockMessageHandler? _announcementHandler;
+  List<Announcement>? _announcements;
+
+  /// Returns a list announcements made by the Flutter framework.
+  List<Announcement>? takeAnnouncements() {
+    assert(inTest);
+    return _announcements;
```

Comment:
I think the semantics of "take" means "remove the returned value from the captured list".

Reply:
Yeah you/re right. I've changed the code to clear out the list afterwards. I don't think we'd need a peak/get function for list. What do you think?